docker-tramp.el

```diff
@@ -123,6 +123,46 @@ to connect to the default user containers."
                  (tramp-remote-shell       "/bin/sh")
                  (tramp-remote-shell-args  ("-i" "-c")))))
 
+;; redefine wait-for-output to fix bug when
+;; trying to access containers which use
+;; busybox's implementation of sh.
+(defun tramp-wait-for-output (proc &optional timeout)
+    "Wait for output from remote command."
+    (unless (buffer-live-p (process-buffer proc))
+      (delete-process proc)
+      (tramp-error proc 'file-error "Process `%s' not available, try again" proc))
+    (with-current-buffer (process-buffer proc)
+      (let* (;; Initially, `tramp-end-of-output' is "#$ ".  There might
+      ;; be leading escape sequences, which must be ignored.
+      (regexp (format "[^#$\n]*%s\r?\\(\\[[0-9;]*[a-zA-Z]*\\)*$" (regexp-quote tramp-end-of-output)))
+      ;; Sometimes, the commands do not return a newline but a
+      ;; null byte before the shell prompt, for example "git
+      ;; ls-files -c -z ...".
+      (regexp1 (format "\\(^\\|\000\\)%s" regexp))
+      (found (tramp-wait-for-regexp proc timeout regexp1)))
+        (if found
```

Comment:
this is a direct copy from `tramp-sh.el`, only the actual regex string is changed. I want to avoid changing anything not relevant to the fix.